main_test.go

```diff
@@ -240,59 +240,91 @@ var _ = Describe("github-review-helper", func() {
 				})
 
 				Context("with a +1 comment", func() {
-					BeforeEach(func() {
-						requestJSON = `{
-  "issue": {
-    "number": 7,
-    "pull_request": {
-      "url": "https://api.github.com/repos/salemove/github-review-helper/pulls/7"
-    }
-  },
-  "comment": {
-    "body": "+1, awesome job!"
-  },
-  "repository": {
-    "name": "github-review-helper",
-    "owner": {
-      "login": "salemove"
-    },
-    "ssh_url": "git@github.com:salemove/github-review-helper.git"
-  }
-}`
-						mockSignature()
-					})
+					var itMarksCommitPeerReviewed = func() {
+						Context("with GitHub request failing", func() {
+							BeforeEach(func() {
+								pullRequests.On("Get", "salemove", "github-review-helper", 7).Return(nil, nil, errors.New("an error"))
+							})
 
-					Context("with GitHub request failing", func() {
-						BeforeEach(func() {
-							pullRequests.On("Get", "salemove", "github-review-helper", 7).Return(nil, nil, errors.New("an error"))
+							It("fails with a gateway error", func() {
+								handle()
+								Expect(responseRecorder.Code).To(Equal(http.StatusBadGateway))
+							})
 						})
 
-						It("fails with a gateway error", func() {
-							handle()
-							Expect(responseRecorder.Code).To(Equal(http.StatusBadGateway))
+						Context("with GitHub request succeeding", func() {
+							BeforeEach(func() {
+								pullRequests.On("Get", "salemove", "github-review-helper", 7).Return(&github.PullRequest{
+									Head: &github.PullRequestBranch{
+										SHA: github.String("1235"),
+										Ref: github.String("feature"),
+									},
+								}, nil, nil)
+							})
+
+							It("reports the status and succeeds", func() {
+								repositories.On("CreateStatus", "salemove", "github-review-helper", "1235", mock.AnythingOfType("*github.RepoStatus")).Return(nil, nil, nil)
+
+								handle()
+
+								Expect(responseRecorder.Code).To(Equal(http.StatusOK))
+								status := repositories.Calls[0].Arguments.Get(3).(*github.RepoStatus)
```

Comment:
I remember writing this (it's just copied for this PR) because I didn't know a good way of doing partial matching of the stubbed methods with the libraries I'm using here. But a quick search now revealed [this feature](https://godoc.org/github.com/stretchr/testify/mock#MatchedBy), which does exactly what I want here, but more in a more elegant way.

I'll update all partial matching to use this instead, but separate from this PR. I've created a task to track this.